### ISSUE TITLE
Control + number mappings don't work

### ISSUE THREAD

**Issue by JulesNP** [neovide/neovide#1646]
**Describe the bug**
Mappings using control and a number (\<C-1>, \<C-2>, etc.) do not work currently, or even register a keypress.

**To Reproduce**
Example using a mapping:
1. Enter `:nnoremap <C-6> <C-^>`
2. Press `Ctrl-6`
3. Nothing happens

No keypress is being registered for any control + number combination:
1. Enter insert mode
2. Press `Ctrl-V`
3. Press any combination of `Ctrl-1`, `Ctrl-2`, etc.
4. Nothing happens

**Expected behavior**
Key combination should be registered and handled by the program.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: Windows 11 (with and without WSL)
 - Neovide Version: 0.10.3
 - Neovim Version: 0.8.0

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**
Log is empty.

**Additional context**
Issue #473 was closed with the solution of manually mapping `<C-6>` to `<C-^>`, but this workaround currently does not work.

**Comment by ZyX-II** [comment]
I have some additional insight: I have modified keyboard_manager.rs according to the diff below and found out that presses like that are shown as

```
[src\window\keyboard_manager.rs:88] key_event = KeyEvent {
    physical_key: ControlLeft,
    logical_key: Control,
    text: None,
    location: Left,
    state: Pressed,
    repeat: false,
    platform_specific: KeyEventExtra {
        text_with_all_modifers: None,
        key_without_modifiers: Control,
    },
}
[src\window\keyboard_manager.rs:88] key_event = KeyEvent {
    physical_key: Digit4,
    logical_key: Character(
        "}",
    ),
    text: None,
    location: Standard,
    state: Pressed,
    repeat: false,
    platform_specific: KeyEventExtra {
        text_with_all_modifers: None,
        key_without_modifiers: Character(
            "}",
        ),
    },
}
```

. This is me pressing `<C-4>`, only using customized version of programming dvorak where it is `<C-}>` (standard Windows US keyboard does not work either, output for `<C-6>` is same, except that "}" is replaced with "6"). In nvim-qt that would have resulted in `<C-]>`. Tracking it further shows that problem here is `text_with_all_modifiers` being `None` when [KeyboardManager::handle_event](https://github.com/neovide/neovide/blob/37f78331232197d6d924f6943d6a4114206264be/src/window/keyboard_manager.rs#L40) [calls](https://github.com/neovide/neovide/blob/37f78331232197d6d924f6943d6a4114206264be/src/window/keyboard_manager.rs#L90) `maybe_get_keybinding` which in absence of dead keys [calls](https://github.com/neovide/neovide/blob/37f78331232197d6d924f6943d6a4114206264be/src/window/keyboard_manager.rs#L145) `text_with_all_modifiers()` which is basically an [accessor function to `platform_specific.text_with_all_modifiers`](https://github.com/neovide/winit/blob/27967af60d408218529255e897964b6cffd94659/src/platform/windows.rs#L310-L312) (present only in neovide fork of winit).
This field is only written in [KeyEvent::create_synthetic](https://github.com/neovide/winit/blob/27967af60d408218529255e897964b6cffd94659/src/platform_impl/windows/keyboard.rs#L426-L447) and [PartialKeyEventInfo::finalize](https://github.com/neovide/winit/blob/27967af60d408218529255e897964b6cffd94659/src/platform_impl/windows/keyboard.rs#L593-L641). After adding adding some more `dbg!()` calls on lines 449 and 641 of this file I found that only `dbg!` on line 641 is triggered.
Further investigation shows that `self.utf16parts` is empty upon [PartialKeyEventInfo::finalize](https://github.com/neovide/winit/blob/27967af60d408218529255e897964b6cffd94659/src/platform_impl/windows/keyboard.rs#L592) call. I have no idea why it is empty, somebody who actually knows winit library should take a look at the issue.

```
diff --git a/src/window/keyboard_manager.rs b/src/window/keyboard_manager.rs
index 72972fb..e862c29 100644
--- a/src/window/keyboard_manager.rs
+++ b/src/window/keyboard_manager.rs
@@ -85,6 +85,7 @@ impl KeyboardManager {
                         let mut next_dead_key = self.prev_dead_key;
                         match input_event {
                             InputEvent::KeyEvent(key_event) => {
+                                dbg!(key_event);
                                 // And a key was pressed
                                 if key_event.state == ElementState::Pressed {
                                     if let Some(keybinding) = self.maybe_get_keybinding(key_event) {
```

**Comment by ZyX-II** [comment]
It appears that `<C-}>` produces `WM_[SYS]KEYDOWN` event and no `WM_[SYS]CHAR` while `<C-q>` which works fine produces both. And only `WM_[SYS]CHAR` [event handler](https://github.com/neovide/winit/blob/27967af60d408218529255e897964b6cffd94659/src/platform_impl/windows/keyboard.rs#L159) populates `utf16parts`.

**Comment by ZyX-II** [comment]
I am wrong where I said that code uses `text_with_all_modifiers`: it does that in presence of dead keys, not in absence. This means that relevant part of code is `KeyEvent.text` computation. It changes the chain, but ultimately does not change the source of the problem: this field of `KeyEvent` is again only computed in `WM_[SYS]CHAR` handler which is not called.

**Comment by fredizzimo** [comment]
These are related as well https://github.com/neovide/neovide/issues/1862 
 and https://github.com/neovide/neovide/issues/994. We have updated the winit version, but something else needs to be done still I think.

**Comment by fredizzimo** [comment]
This should be fixed by #1899, it would be great if you can test that.
